### PR TITLE
[DNM]Revert "Rebases to /vg/station13 code."

### PR DESCRIPTION
**Pull request by MrsTonedOne**
Reverts tgstation/tgstation#23087

Memleaking, crashing the servers when it goes over the limit on 32bit process

@PJB3005 

DNM - This is mainly to manual merge on the servers to stop the crashes.

Once we get a general idea on how long until a bug fix from byond we can decide on if we want to merge this. I have a feeling that there will be some work around lummox will find for us in the meantime.

Related byond bug report http://www.byond.com/forum/?post=2220974

### CODE REVIEW

**Comment by PJB3005** [comment]
> memleaking

Can we get Lummox on the line.

**Comment by MrStonedOne** [comment]
i'm more likely to assume its that you are allocating something somewhere and keeping references/not qdeling it.

not for anything against you, but because out of 10 memory leaks i've found in ss13, only 2 ended up being byonds fault and i don't want to bug lummox until i have a better idea since he gets mad when he has to debug our code.

**Comment by PKPenguin321** [comment]
Not like this :cry:

**Comment by Durkel** [comment]
Damn, I was looking forward to this too.

**Comment by MrStonedOne** [comment]
I'm investigating the leak, i'll see if i can figure it out.

**Comment by Allfd** [comment]
We were able to "fix" this on Paradise with this PR https://github.com/ParadiseSS13/Paradise/pull/6475

**Comment by MrStonedOne** [comment]
So wait, its confirmed to exist between rounds?

That's a valuable hint.

Sadly we've done a lot of work on keeping the between round time smooth and user friendly. A shutdown/reconnect would ruin that, and i'd like to avoid that.

But I should be able to figure it out with that hint.

**Comment by GunHog** [comment]
I really hope you can fix it :cry:

**Comment by MrStonedOne** [comment]
update, I have confirmed this is a byond bug, the memory remains after the server reboots.

it's the color matrix.

File: code/modules/lighting/lighting_overlay.dm [hunk 1]
```diff
@@ -1,117 +0,0 @@
...
-	var/datum/lighting_corner/cg  = T.corners[2] || dummy_lighting_corner
-	var/datum/lighting_corner/cb  = T.corners[4] || dummy_lighting_corner
-	var/datum/lighting_corner/ca  = T.corners[1] || dummy_lighting_corner
-
-	var/max = max(cr.cache_mx, cg.cache_mx, cb.cache_mx, ca.cache_mx)
-
-	color  = list(
-		cr.cache_r, cr.cache_g, cr.cache_b, 0,
-		cg.cache_r, cg.cache_g, cg.cache_b, 0,
-		cb.cache_r, cb.cache_g, cb.cache_b, 0,
```
Comment (MrStonedOne) [inline code/modules/lighting/lighting_overlay.dm]: commenting out this stops the memory leak. =\

**Comment by GunHog** [comment]
So why not just disable colored lighting for now?

**Comment by MrStonedOne** [comment]
nope, vg lighting uses color matrixs for the shaded/faded look the corners have. It's the entire trick behind how that works.

**Comment by GunHog** [comment]
So then to disable that code means to remove the reason why it exists in the first place...I wish it were something as simple as clearing that cache or dumping whatever color stuff is doing this. I wish you the best of luck, MSO! I really hate to see this fail :(

**Comment by Allfd** [comment]
@MrStonedOne 

Memory leak

I can send you my full report on this over discord, we were able to track the memory leak across rounds.

**Comment by Krausus** [comment]
> Sadly we've done a lot of work on keeping the between round time smooth and user friendly. A shutdown/reconnect would ruin that, and i'd like to avoid that.

Well, it's not what I'd call a *good* solution, but forcibly terminating DD *does* make clients automatically reconnect, since their connection drops out unexpectedly.

There's probably some cleaner way to make clients reconnect, perhaps using Javascript, but murdering DD seems good enough.

**Comment by PJB3005** [comment]
> it's the color matrix

*knew it*

**Comment by StarSmasher44** [comment]
Couldn't you just set the cause of the memory leak (in this case I assume it's the color variable) to null before doing a reboot or would that still make no difference?
Or possibly deleting the lighting overlays from the world before round end. Just throwing out random ideas since this isn't my strong suit

**Comment by MrStonedOne** [comment]
@StarSmasher44 i just tried that, it made it worse actually.

**Comment by StarSmasher44** [comment]
Well I didn't expect that to happen honestly, but I'd be leaning more towards this being a byond bug rather than something ss13 related. Perhaps if you'd tell lumox whats up and what you believe tje cause is he may be able to confirm it or learn us something useful in the process

**Comment by MrStonedOne** [comment]
>but I'd be leaning more towards this being a byond bug

Oh ya, anything causing memory to leak around after reboot is 100% a byond bug.

**Comment by MrStonedOne** [comment]
I was able to reproduce the bug

by that, i mean i was able to port enough of our procs/framework to get the mc working AND get lighting to work in a blank project, and it happened there too.

http://www.byond.com/forum/?post=2220974

**Comment by Cyberboss** [comment]
Servers not updated since friday, #24673 still a major problem :(

**Comment by KorPhaeron** [comment]
@tgstation/commit-access 

I know MSO is working on a fix for this but continuously keeping this test merged while the game is otherwise unplayable doesn't really make sense as far as downstreams go. I think we should revert and then readd this when it works.

**Comment by Cyberboss** [comment]
Seriously, I love /vg/ lights but I feel like this is holding back all of /tg/ at the moment.

**Comment by lzimann** [comment]
This will probably conflict soon... Also we have no idea how long will take to Lummox to fix the leak issue.

**Comment by PJB3005** [comment]
Just merge it. My code's haunted and we need Lummox to fix it.

**Comment by AnturK** [comment]
Let's give Lummox 24h at least.

**Comment by MrStonedOne** [comment]
ya, lummox saw the thread last night, and just started working on it today, just keep an eye on that thread.

**Comment by Cyberboss** [comment]
It's been resolved, this can be closed once the servers update byond

**Pull request closed without merging**